src/entry.c

```diff
@@ -1,39 +1,60 @@
-#include <stdbool.h>
 #include "server.h"
 #include "serverassert.h"
 #include "entry.h"
 
-#include <stdbool.h>
-
 /*-----------------------------------------------------------------------------
- * Entry API
+ * Entry Implementation
  *----------------------------------------------------------------------------*/
 
-/* The entry pointer is the field sds. We encode the entry layout type
- * in the field SDS header. Field type SDS_TYPE_5 doesn't have any spare bits to
- * encode this so we use it only for the first layout type.
+/* There are 3 different formats for the "entry".  In all cases, the "entry" pointer points into the
```

Comment:
there are 4 right? expiration existence only multiply the past 2 encoding types (embedded value and non-embedded value)

Reply:
In types 2 and 3, the expiration is optional.  I didn't consider/document these as fundamentally different types.  If we do that, there would be 5 types.